### DISCUSSION TITLE
WebView on OSX size to fit window?

### DISCUSSION THREAD

**Discussion by hoffenmerCode** [beeware/toga#1305] [Q&A]
Writing my own HTML code to be displayed by toga inside a Box which is part of a splitContainer. 
The code below results in the 'manually' fitting the HTML file to the exact size of the box (500,500).
When I remove the width and height definition below the HTML view is tiny (50,50) by default. 
I tried using flex=1, doesn't work. 
Is there a way to define a "Box" and have its content fill the box 100% without needing to define the actual width/height?

`html_view = toga.WebView(url="file:///Users/******/Projects/Beeware/MaSE/sample_html.html", style=Pack(flex=1, width=500, height=500))
        containerE_html = toga.Box(style=Pack(flex=1), children=[
            html_view
        ])

        containerE = toga.OptionContainer()
        containerE.add("Native", containerE_native)
        containerE.add("HTML", containerE_html)`

**Comment by freakboy3742** [comment]
What you're requesting isn't possible. Toga's layout algorithm doesn't take the content of the Webview into account when computing the size of the webview - because the web content itself will be laid out based on the amount of space the web view provides. This is a fundamental property of HTML, so it's not something Toga is in a position to work around, either.

So - you need to specify a size for the webview. Toga will provide a minimum size (which you can override with min_width and min_height style directives); but fundamentally you need to provide the size.

**Reply by hoffenmerCode** [reply]
Thanks for the quick reply. 

1. On the same token, table rows. If I have text that overflows the width of the row, can we somehow wrap the text,  or expand the height of the row or make it scrollable? At the moment, oversized text just sort of hangs centered - with the top/bottom cut off. I know I asked a similar question before - declaring row height, I know it cannot be done, I just wonder if toga can 'wrap' text as needed to fit the text inside the row instead of just declaring row height for the sake of having custom row heights. I just need the text to fit. Or use another solution to do the same? 

2. Can you contact me directly at 858-353-2003 (text please). I run a small business that uses Toga now and I would like to discuss something with you please.

**Reply by freakboy3742** [reply]
1. That's not currently an option. Tables are designed for short form summary text, not long form text. I'm not sure I have any good options for you in the current code, other than a "detail view" that displays long content in a separate panel based on the current table selection.

2. I'm not based in the US, so SMS text messages aren't a viable option; however, if you email russell@keith-magee.com I'm happy to discuss anything commercially sensitive.

**Reply by hoffenmerCode** [reply]
2. Emailed from hoffenmer - sometimes goes to spam
